module_utils/ansible_nailgun_cement.py

```diff
@@ -476,6 +477,13 @@ def find_compute_profile(module, name, failsafe=False):
     return handle_find_response(module, compute_profile, message="No compute profile found for %s" % name, failsafe=failsafe)
 
 
+def find_compute_attribute(module, compute_profile_name, compute_resource_name, failsafe=False):
+    search = {'search': 'compute_profile="{}" and compute_resource="{}"'.format(compute_profile_name, compute_resource_name)}
+    compute_attribute = ComputeAttribute().search(set(), search)
+    return handle_find_response(module, compute_attribute, message="No compute attribute found for %s %s" % (compute_profile_name, compute_resource_name),
```

Comment:
Also it is possible this can return more one attribute? I think it looks like it could, a profile and resource can have many attributes, right? This will cause an error I think https://github.com/theforeman/foreman-ansible-modules/blob/master/module_utils/ansible_nailgun_cement.py#L616

Reply:
I think this not possible because of constraints in foreman https://github.com/theforeman/foreman/blob/develop/app/models/compute_attribute.rb#L7. You can only have 1 compute_attribute per tuple compute_resource/compute_profile